### ISSUE TITLE
Unable to connect to VoiceChannel due to userlimit when already connected

### ISSUE THREAD

**Issue by Frederikam** [DV8FromTheWorld/JDA#305]
Since https://github.com/DV8FromTheWorld/JDA/commit/e77a49819a3e5ea51aa1f3e0e71149d52102ea5d I have been getting this exception calling `AudioManager#openAudioConnection()` where the following conditions are true:

1. The bot is already in the VoiceChannel
2. The VoiceChannel is at or above the user cap

```
net.dv8tion.jda.core.exceptions.PermissionException: Unable to connect to VoiceChannel due to userlimit! Requires permission MANAGE_CHANNEL to bypass
    net.dv8tion.jda.core.managers.impl.AudioManagerImpl.openAudioConnection(AudioManagerImpl.java:91)
```

I believe the intended behaviour is for this to silently return, though I could easily add my own check for this to prevent this exception from happening.

**Comment by DV8FromTheWorld** [comment]
Yup, this is a bug related to the order of checking.

**Comment by DV8FromTheWorld** [comment]
Fixed as of df2a5e5
